### PR TITLE
feat: Add `aws-ruby-step-functions` example

### PR DESCRIPTION
**Pull request by Pigius**
# Big picture behind

Currently, there aren't many examples of using the serverless approach with Ruby for AWS. Ruby is a great and mature language with a great community and lots of gems.

# General message

There are not many examples of Step Functions, and some of the existing ones do not use the Step Functions plugin, but have a ready-made schema that you can copy.

Luckily we also now have Workflow Studio for AWS Step Functions, so it's easier to design state machines.

Diagrams could be found here:

step-functions-diagram

step-functions-detailed

Extra points:
- There is an issue with using `sls` variables with this plugin, there are already issues for that.
- Used the Ruby.27 runtime, as for the Ruby2.5 will be End of Support at the end of August.

### CODE REVIEW

**Comment by pgrzesik** [comment]
Thanks @Pigius :raised_hands: As discussed, it's temporarily paused until https://github.com/serverless-operations/serverless-step-functions/pull/451 is merged and new version of the plugin is released, due to incompatibility with new variable resolver.

**Comment by Pigius** [comment]
@pgrzesik the package for the step function was updated to the newest version. Everything works seamlessly. Ready to be checked by you 🚀

File: README.md [hunk 1]
```diff
@@ -13,14 +13,12 @@ A collection of ready-to-deploy [Serverless Framework](https://github.com/server
...
 
-- [Serverless Examples](#serverless-examples)
-  - [Table of Contents](#table-of-contents)
-  - [Getting Started](#getting-started)
-  - [Examples](#examples)
-  - [Community Examples](#community-examples)
-  - [Contributing](#contributing)
-    - [Adding example code](#adding-example-code)
-    - [Adding a community example](#adding-a-community-example)
+- [Getting Started](#getting-started)
```
Comment (pgrzesik) [inline README.md:16]: I'm guessing this should not be a part of this PR, right? :D

**Review by pgrzesik — COMMENTED** [review]
It looks great, thank you @Pigius :bow: I have only one question about seemingly unrelated change that landed in this PR and we should be good to go :100:

File: README.md (hunk 1 above)
Comment (Pigius) [reply README.md]: @pgrzesik fixed. Not sure what it changed, I've run npm run docs and this happened :/ But fixed anyway :)

Comment (pgrzesik) [reply README.md]: Interesting, I'll look into it :100:

**Review by pgrzesik — APPROVED** [review]
Thank you for awesome, as always, contribution @Pigius :bow:

**Pull request merged**